### ISSUE TITLE
Enable / Disable ordered target statistic encoder

### ISSUE THREAD

**Issue by david-waterworth** [catboost/catboost#894]
I understand I can enable/disable the order boosting using the `boosting_type` argument. How does this impact the ordered target statistic encoder? Do you always perform ordered target statistic encoding even if boosting_type=='plain'? is there a way of switching from ordered to plain target statistic encoding?

I'd like to run a series of experiments with these innovations on and off. My hypothesis is without enough data the model is worse with ordered boosting

**Comment by annaveronika** [comment]
It does not impact ordered statistics, we always perform it. There is not switch to plain target statistic encoder, we don't have it, because it usually leads to overfitting.
Ordered boosting works well for smaller data, because it helps to prevent overfitting. So we use it only for small datasets.
You can look on some experiments in https://arxiv.org/abs/1706.09516

**Comment by david-waterworth** [comment]
Thanks @annaveronika

I'm fitting a smallish dataset (1 year * 96 samples per day) with strong dependency structure. To cross-validate, I use blocked cross-validation (7-day blocks). The CV metric is CV(RMSE) which is the standard RMSE scaled by the mean of the target. A secondary metric NMBE is used - NMBE is normalised mean bias error, this should be close to zero.

`np.mean(y - model.predict(X)) / np.mean(X)`

I need to do more experiments but my observations to date have been that the NMBE is quite a lot higher for ordered boosting/ordered statistics. Perhaps because of the dependency structure and the use of blocked cross-validation results in a model which is quite shallow and not may trees?

**Comment by david-waterworth** [comment]
Here's a typical example. I'm using hyperopt to cross-validate. I've set up my pipeline so I can either use catboost to encode the categorical features or use ordered target encoding.

The chart rows are the cross-validation results for ordered and plain boosting, the columns are the test and train metrics. I'm optimising CV(RMSE) and calculating NMBE. The colours show ordered (grey) and plain (orange) target encoding.

Ordered target encoding results in non-zero NMBE for the training set, the test set results are better it's hard to argue one is better than another.

Ordered boosting does seem to result in a slightly better model.

![image](https://user-images.githubusercontent.com/5028974/60777329-368d8280-a175-11e9-8b12-9c67eb49e6be.png)
